### ISSUE TITLE
Problem with initial data load

### ISSUE THREAD

**Issue by attzonko** [attzonko/mattermost_bot#7]
@Sebastien-Meiffren Reported the following issue:

```
Traceback (most recent call last):
  File "C:/Users/chuiv/PycharmProjects/bot_mattermost/run.py", line 5, in <module>
    Bot().run()
  File "C:\Users\chuiv\PycharmProjects\bot_mattermost\mattermost_bot\bot.py", line 29, in __init__
    settings.SSL_VERIFY)
  File "C:\Users\chuiv\PycharmProjects\bot_mattermost\mattermost_bot\mattermost_v4.py", line 110, in __init__
    self.login(team, email, password)
  File "C:\Users\chuiv\PycharmProjects\bot_mattermost\mattermost_bot\mattermost.py", line 184, in login
    self.user = self.api.login(team, email, password)
  File "C:\Users\chuiv\PycharmProjects\bot_mattermost\mattermost_bot\mattermost_v4.py", line 25, in login
    self.load_initial_data()
  File "C:\Users\chuiv\PycharmProjects\bot_mattermost\mattermost_bot\mattermost_v4.py", line 34, in load_initial_data
    self.default_team_id = self.teams[0]['id']
IndexError: list index out of range
```
If I print `self.teams = self.get('/teams')`, I get an empty array []

If I connect to the webclient like a normal person / user, everything works fine.

**Comment by attzonko** [comment]
@Sebastien-Meiffren Can you give some more information?

1. What version of the Mattermost server are you running?
2. Have you disabled the v3 APIs completely?
3. In the mattermost_bot settings what is the BOT_URL set to?
4. What about the MATTERMOST_API_VERSION?

**Comment by Sebastien-Meiffren** [comment]
Hi,

1. I didn't do the install, but it is a new version (2 weeks old)
2. Dunno. How can I check that in the console ? => probably cause when I query the v3 api address, I've got 404
3. To my URL with the v4 api address 
4. 4 (I used your conf as an example)

**Comment by attzonko** [comment]
On for the first one, you can check by going through the 'three bar' menu and selecting 'About Mattermost'

It sounds like your instance is not returning any teams from the '/teams' v4 API. So either the login failed, or there is another issue, or the bot user you are using is not part of any teams yet?

**Comment by Sebastien-Meiffren** [comment]
Hi,

1. `Version: 4.6.0 Database: postgres`

I do have a team selected as I can login properly with the web browser and I can chat in a channel in a specific team

I don't know if it helps, but I'm on windows ...

**Comment by seLain** [comment]
@Sebastien-Meiffren  Is the specific team an open team or private team ?

**Comment by Sebastien-Meiffren** [comment]
I'm not sure what you mean by private or open team but no everyone can enter this team. They need to be invited.
The user for the bot has been invited

**Comment by seLain** [comment]
That's a private team. Could you try to set that team open for everyone to join, and test your run.py again ?

I just found an issue #8 associated with getting all teams at login, which might be related to this problem.

**Comment by attzonko** [comment]
I believe this issue should be resolved by the commit @seLain made #8
